### PR TITLE
Add support for mlint, a MATLAB linter

### PR DESCRIPTION
**Pull request by awlayton**
I had to redirect stderr to stdout, so this probably doesn't work in Windows. Let  me know if I should do it a different way. I looked some and didn't see current support for any other linters that output to stderr.

### CODE REVIEW

File: ale_linters/matlab/mlint.vim [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            continue
+        endif
+
+        let l:lnum = l:match[1] + 0
+        let l:col = l:match[2] + 0
+        let l:code = l:match[3]
+        let l:text = l:match[4]
+
+        " Suppress erroneous waring about file name
+        if l:code ==# 'FNDEF'
```
Comment (w0rp) [inline ale_linters/matlab/mlint.vim]: We might be able to turn this on in future. There is some support for copying the same filename to the temporary directory to come eventually.

File: ale_linters/matlab/mlint.vim [hunk 2]
```diff
@@ -0,0 +1,59 @@
+" Author: awlayton <alex@layton.in>
+" Description: mlint for MATLAB files
+
+if exists('g:loaded_ale_linters_matlab_mlint')
```
Comment (w0rp) [inline ale_linters/matlab/mlint.vim]: We don't need the `exists` guards any more. They are being removed to aid some testing, and we should only be loading these files at most once per filetype anyway, I believe.

**Comment by awlayton** [comment]
Thanks for pointing out the right way to handle stderr. I think I addressed all your changes @w0rp.

File: ale_linters/matlab/mlint.vim [hunk 3]
```diff
@@ -0,0 +1,57 @@
+" Author: awlayton <alex@layton.in>
+" Description: mlint for MATLAB files
+
+let g:loaded_ale_linters_matlab_mlint = 1
```
Comment (w0rp) [inline ale_linters/matlab/mlint.vim]: You left the "loaded" flag lying around.

**Comment by w0rp** [comment]
Looks good to me. Cheers! :beers:

**Pull request merged**